runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -780,19 +791,24 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
 
    if (abortCompilation)
       {
+      enterPerClientAllocationRegion();
       if (clientOptions)
          {
          TR_Memory::jitPersistentFree(clientOptions);
```

Comment:
Maybe we should specify the persistent memory to be used here as we know it's the one stored in the session:
`getClientData()->persistentMemory()->freePersistentMemory(clientOptions)`
`getClientData()` must exist if clientOptions have been created

Reply:
Done.